Put routepage into a sub folder

vesameskanen opened the pull request:
- Route page and its sub components moved to a sub folder 'routepage'
- Some components renamed, dead ones removed
- Some runtime proptype warnings fixed
- Moment library use reduced
- Timestore exposes unix time instead of cloned moment object, which changes at every access

vesameskanen committed:
chore: remove dead component

vesameskanen committed:
fix: remove dead scss

vesameskanen committed:
chore: move route page components to dedicated folder

vesameskanen committed:
chore: use shorter names, update unit tests

vesameskanen committed:
fix: feed is not mandatory for AlertRow

vesameskanen committed:
fix: MenuDrawer does not require className

vesameskanen committed:
chore: remove moment from RouteAlertsContainer.js

vesameskanen committed:
Merge branch 'v3' into techdebt2

vesameskanen committed:
fix: use consistent version of react-autosuggest everywhere

vesameskanen committed:
chore: update lockfile

vesameskanen committed:
Merge branch 'v3' into techdebt2

vesameskanen committed:
fix: mark serviceDay and scheduledDeparture mandatory, add missing unit test mock data

vesameskanen committed:
fix: refactor time store

It does not make sense to pass current time as moment object.
99% of users convert it to unix time immediately.

vesameskanen committed:
fix: remove unused emit parameter

vesameskanen committed:
fix: assume unix time in StopListContainers

vesameskanen committed:
feat: unix to YYYMMDD

vesameskanen committed:
fix: route page uses new unix current time

vesameskanen committed:
fix: all remaining components to use unix current time

vesameskanen committed:
chore: update tests

vesameskanen committed:
fix: add missing config for unix time conversion

vesameskanen committed:
feat: unit time util without moment

vesameskanen committed:
chore: use new unix time util

vesameskanen committed:
chore: less moment

vesameskanen committed:
chore: less moment

vesameskanen committed:
fix: StopAlerts unit tests (mockContext added)

vesameskanen committed:
chore: less moment

vesameskanen committed:
chore: less moment

vesameskanen committed:
Merge branch 'v3' into techdebt2

optionsome commented on app/component/routepage/TripStopListContainer.js:
```diff
@@ -48,7 +47,7 @@ class TripStopListContainer extends React.PureComponent {
     const vehicles = groupBy(
       values(propVehicles).filter(
-        vehicle => currentTime - vehicle.timestamp * 1000 < 5 * 60 * 1000,
+        vehicle => currentTime - vehicle.timestamp < 5 * 60,
```
In some places the timestamp is still multiplied by 1000 and in some places it isn't. Is the timestamp itself edited in this pr or the currentTime?

vesameskanen replied:
Vehicle timestamps are unix time i.e. seconds after epoch. currentTime used to be a moment object but this PR indeed changes it to unix time.

vesameskanen committed:
chore: rename component to something meaningful

vesameskanen committed:
fix: don't call time parameter as trip

vesameskanen committed:
fix: don't repeatedly test if trip is null, it is a required prop

vesameskanen committed:
fix: don't render trip popup if trip is not known

vesameskanen committed:
fix: trip is not always known in TripLink

vesameskanen committed:
fix: stopcard description prop is not always defined

vesameskanen committed:
chore: remove test which verifies that component renders with invalid props

vesameskanen committed:
fix: apply sensible filtering for geometry

Checking that point.lat !== null did not break rendering because point.lat === undefined :)
